src/linear_algebra/determinant.lean

```diff
@@ -40,7 +40,8 @@ open equiv equiv.perm finset function
 namespace matrix
 open_locale matrix big_operators
 
-variables {n : Type u} [decidable_eq n] [fintype n] {R : Type v} [comm_ring R]
+variables {m n : Type u} [decidable_eq n] [fintype n] [decidable_eq m] [fintype m] {R : Type v}
+  [comm_ring R]
```

Comment:
I prefer to start a new `variables` statement if it becomes longer than one line. That way it's easier to group related declarations. Also @eric-wieser's suggestion of not repeating the universe parameter.
```suggestion
variables {m n : Type*} [decidable_eq n] [fintype n] [decidable_eq m] [fintype m]
variables {R : Type v} [comm_ring R]
```